src/algebra/zset.rs

```diff
@@ -301,14 +301,14 @@ fn checked_zset_integer_weights_tests() {
 
 #[cfg(test)]
 #[derive(Debug, Clone, Hash, Eq, PartialEq)]
-struct TestTuple {
-    left: i64,
-    right: i64,
+pub(crate) struct TestTuple {
+    pub(crate) left: i64,
+    pub(crate) right: i64,
 }
 
 #[cfg(test)]
 impl TestTuple {
-    fn new(left: i64, right: i64) -> Self {
+    pub fn new(left: i64, right: i64) -> Self {
```

Comment:
I suggest moving all tests into a module (within the same file), so you only need to write one `#[cfg(test)]` annotation for the whole module. 

From https://doc.rust-lang.org/book/ch11-03-test-organization.html:
```
The convention is to create a module named tests in each file to contain the test functions and to annotate the module with cfg(test).
```